service/character.go

```
@@ -0,0 +1,140 @@
+	// TODO: Implement?
+	if url == "" {
+		return nil, fmt.Errorf("Empty url")
+	}
+
+	if file == "" {
+		return nil, fmt.Errorf("Empty file path")
+	}
+
+	return &Service{url, file}, nil
+}
+
+func (s *Service) GetCharacters() ([]model.Character, error) {
+	// TODO: Implement
+	csvFile, err := os.Open(s.file)
+	if err != nil {
+		log.Fatal("Error creating file reader", err)
+	}
+	r := csv.NewReader(csvFile)
+	defer csvFile.Close()
+
+	var character model.Character
+	var characters []model.Character
+
+	for {
+		record, err := r.Read()
+
+		if err == io.EOF {
+			break
+		}
+
+		charID, _ := strconv.ParseInt(record[0], 10, 64)
+
+		character.ID = int(charID)
+		character.Name = record[1]
+		character.Status = record[2]
+		character.Species = record[3]
+		character.Gender = record[4]
+
+		characters = append(characters, character)
+	}
+
+	return characters, nil
+}
+
+func (s *Service) GetCharacterByID(id int) (*model.Character, error) {
+	// TODO: Implementa
+	csvFile, err := os.Open(s.file)
+	if err != nil {
+		log.Fatal("Error creating file reader", err)
+	}
+	r := csv.NewReader(csvFile)
+	defer csvFile.Close()
+
+	var character model.Character
+
+	for {
+		record, err := r.Read()
+
+		if err == io.EOF {
+			break
+		}
+
+		charID, _ := strconv.ParseInt(record[0], 10, 64)
+		fileId := int(charID)
+
+		if fileId == id {
+			fmt.Println("character found")
+			character.ID = fileId
+			character.Name = record[1]
+			character.Status = record[2]
+			character.Species = record[3]
+			character.Gender = record[4]
+			break
+		}
+	}
+
+	return &character, nil
+}
+
+func (s *Service) InsertExternalCharacter(id int) error {
+	reqUrl := s.url + strconv.Itoa(id)
+
+	csvW, err := os.OpenFile(s.file, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	if err != nil {
+		log.Fatal("Error creating file writer")
+	}
+	w := csv.NewWriter(csvW)
+
+	defer csvW.Close()
+
+	res, err := http.Get(reqUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	data, _ := ioutil.ReadAll(res.Body)
```

You should never ignore errors. What happens if an error occurs here?